apps/commudle-admin/src/app/services/stats/stats-events.service.ts

```diff
@@ -6,66 +6,51 @@ import { ApiRoutesService } from 'apps/shared-services/api-routes.service';
-  ) { }
-
-
+  constructor(private http: HttpClient, private apiRoutesService: ApiRoutesService) {}
 
   uniqueVisitors(eventId): Observable<any> {
-    let params = new HttpParams().set('event_id', eventId);
-    return this.http.get<any>(
-      this.apiRoutesService.getRoute(API_ROUTES.STATS.EVENTS.UNIQUE_VISITORS), { params }
-    );
+    const params = new HttpParams().set('event_id', eventId);
+    return this.http.get<any>(this.apiRoutesService.getRoute(API_ROUTES.STATS.EVENTS.UNIQUE_VISITORS), { params });
   }
 
   customRegistration(eventId): Observable<any> {
-    let params = new HttpParams().set('event_id', eventId);
-    return this.http.get<any>(
-      this.apiRoutesService.getRoute(API_ROUTES.STATS.EVENTS.CUSTOM_REGISTRATION), { params }
-    );
+    const params = new HttpParams().set('event_id', eventId);
+    return this.http.get<any>(this.apiRoutesService.getRoute(API_ROUTES.STATS.EVENTS.CUSTOM_REGISTRATION), { params });
   }
 
-
   simpleEventRegistration(eventId): Observable<any> {
-    let params = new HttpParams().set('event_id', eventId);
-    return this.http.get<any>(
-      this.apiRoutesService.getRoute(API_ROUTES.STATS.EVENTS.SIMPLE_EVENT_REGISTRATION), { params }
-    );
+    const params = new HttpParams().set('event_id', eventId);
+    return this.http.get<any>(this.apiRoutesService.getRoute(API_ROUTES.STATS.EVENTS.SIMPLE_EVENT_REGISTRATION), {
+      params,
+    });
   }
 
-
   attendees(eventId): Observable<any> {
-    let params = new HttpParams().set('event_id', eventId);
-    return this.http.get<any>(
-      this.apiRoutesService.getRoute(API_ROUTES.STATS.EVENTS.ATTENDEES), { params }
-    );
+    const params = new HttpParams().set('event_id', eventId);
+    return this.http.get<any>(this.apiRoutesService.getRoute(API_ROUTES.STATS.EVENTS.ATTENDEES), { params });
   }
 
-
-
   discussions(eventId): Observable<IDiscussions> {
-    let params = new HttpParams().set('event_id', eventId);
-    return this.http.get<IDiscussions>(
-      this.apiRoutesService.getRoute(API_ROUTES.STATS.EVENTS.DISCUSSIONS), { params }
-    );
+    const params = new HttpParams().set('event_id', eventId);
+    return this.http.get<IDiscussions>(this.apiRoutesService.getRoute(API_ROUTES.STATS.EVENTS.DISCUSSIONS), { params });
   }
 
-
   polls(eventId): Observable<IPolls> {
-    let params = new HttpParams().set('event_id', eventId);
-    return this.http.get<IPolls>(
-      this.apiRoutesService.getRoute(API_ROUTES.STATS.EVENTS.POLLS), { params }
-    );
+    const params = new HttpParams().set('event_id', eventId);
+    return this.http.get<IPolls>(this.apiRoutesService.getRoute(API_ROUTES.STATS.EVENTS.POLLS), { params });
   }
 
+  speakers(eventId): Observable<any> {
+    const params = new HttpParams().set('event_id', eventId);
+    return this.http.get<any>(this.apiRoutesService.getRoute(API_ROUTES.STATS.EVENTS.SPEAKERS), { params });
+  }
 
+  memberStats(eventId): Observable<any> {
```

Comment:
Add interface for each stat. Stat interfaces can be named as IStat<Name> and put in a folder